notificationdemo/source/android/app/src/main/java/org/kaaproject/kaa/demo/notification/kaa/KaaManager.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright 2014-2016 CyberVision, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kaaproject.kaa.demo.notification.kaa;
+
+import android.app.Activity;
+import android.util.Log;
+
+import org.kaaproject.kaa.client.SimpleKaaClientStateListener;
+import org.kaaproject.kaa.demo.notification.util.NotificationConstants;
+import org.kaaproject.kaa.client.AndroidKaaPlatformContext;
+import org.kaaproject.kaa.client.Kaa;
+import org.kaaproject.kaa.client.KaaClient;
+import org.kaaproject.kaa.client.notification.NotificationListener;
+import org.kaaproject.kaa.client.notification.NotificationTopicListListener;
+import org.kaaproject.kaa.client.notification.UnavailableTopicException;
+import org.kaaproject.kaa.common.endpoint.gen.Topic;
+
+import java.util.List;
+
+/**
+ * Performs initialization of application resources including initialization of the Kaa client.
+ * Handles the Kaa client lifecycle.
+ *
+ * @see <a href="http://docs.kaaproject.org/display/KAA/Using+notifications">Programming guide</a>
+ */
+public class KaaManager {
```

Comment:
This class contains very primitive functionality and code may look more clear if you will move this code to main activity. This is not "production" approach, but simplifies code review for Kaa newbies